uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/providers/IdentityProviderEndpointsAliasMockMvcTests.java

```diff
@@ -590,158 +739,268 @@ protected AliasFeatureDisabled() {
+                }
             }
 
-            private void shouldReject_OtherPropertiesChangedWhileAliasPropertiesUnchanged(
-                    final IdentityZone zone1,
-                    final IdentityZone zone2
-            ) throws Throwable {
-                final IdentityProvider<?> originalIdp = executeWithTemporarilyEnabledAliasFeature(
-                        aliasFeatureEnabled,
-                        () -> createIdpWithAlias(zone1, zone2)
-                );
+            /**
+             * Test handling of IdPs with an existing alias when the alias feature is now switched off.
+             */
+            @Nested
+            class ExistingAlias {
+                @Test
+                void shouldReject_OtherPropertiesChangedWhileAliasPropertiesUnchanged_UaaToCustomZone() throws Throwable {
+                    shouldReject_OtherPropertiesChangedWhileAliasPropertiesUnchanged(IdentityZone.getUaa(), customZone);
+                }
 
-                // change non-alias property without setting alias properties to null
-                originalIdp.setName("some-new-name");
-                shouldRejectUpdate(zone1, originalIdp, HttpStatus.UNPROCESSABLE_ENTITY);
-            }
+                @Test
+                void shouldReject_OtherPropertiesChangedWhileAliasPropertiesUnchanged_CustomToUaaZone() throws Throwable {
+                    shouldReject_OtherPropertiesChangedWhileAliasPropertiesUnchanged(customZone, IdentityZone.getUaa());
+                }
 
-            @Test
-            void shouldAccept_SetOnlyAliasPropertiesToNull_UaaToCustomZone() throws Throwable {
-                shouldAccept_SetOnlyAliasPropertiesToNull(IdentityZone.getUaa(), customZone);
-            }
+                private void shouldReject_OtherPropertiesChangedWhileAliasPropertiesUnchanged(
+                        final IdentityZone zone1,
+                        final IdentityZone zone2
+                ) throws Throwable {
+                    final IdentityProvider<?> originalIdp = executeWithTemporarilyEnabledAliasFeature(
+                            aliasFeatureEnabled,
+                            () -> createIdpWithAlias(zone1, zone2)
+                    );
+
+                    // change non-alias property without setting alias properties to null
+                    originalIdp.setName("some-new-name");
+                    shouldRejectUpdate(zone1, originalIdp, HttpStatus.UNPROCESSABLE_ENTITY);
+                }
 
-            @Test
-            void shouldAccept_SetOnlyAliasPropertiesToNull_CustomToUaaZone() throws Throwable {
-                shouldAccept_SetOnlyAliasPropertiesToNull(customZone, IdentityZone.getUaa());
-            }
+                @Test
+                void shouldAccept_SetOnlyAliasPropertiesToNull_UaaToCustomZone() throws Throwable {
+                    shouldAccept_SetOnlyAliasPropertiesToNull(IdentityZone.getUaa(), customZone);
+                }
```

Comment:
Looks good. Leaving comments to track my progress as this is challenging to read through.